eng/scripts/Language-Settings.ps1

```diff
@@ -1,41 +1,51 @@
 $Language = "go"
 $packagePattern = "go.mod"
-# rewrite from ChangeLog-Operations.ps1 used in Get-ChangeLogEntriesFromContent for go uses vx.x.x as version number
-$RELEASE_TITLE_REGEX = "(?<releaseNoteTitle>^\#+\s+(?<version>v$([AzureEngSemanticVersion]::SEMVER_REGEX))(\s+(?<releaseStatus>\(.+\))))"
+$LanguageDisplayName = "go"
+$VERSION_LINE_REGEX = ".+\s*=\s*`".*v?(?<version>$([AzureEngSemanticVersion]::SEMVER_REGEX))`""
 
-# rewrite from artifact-metadata-parsing.ps1 used in RetrievePackages for fetch go single module info
-function Get-go-PackageInfoFromPackageFile ($pkg, $workingDirectory)
+function Get-GoPackageId ($pkgDirectory) 
 {
-    $workFolder = $pkg.Directory
-    $releaseNotes = ""
+    $serviceDir = ""
+    $pkgId = ""
+    $sdkType = ""
 
-    if ($workFolder -match "sdk[\/|\\]((?<repopath>(?<service>.*?)[\/|\\])?(?<arm>arm)?(?<pkgname>.*))")
+    if ($pkgDirectory -match "sdk[\/|\\]((?<repopath>(?<service>.*?)[\/|\\])?(?<arm>arm)?(?<pkgname>.*))")
     {
         if ($matches["arm"])
         {
             $packageName = "arm" + $matches["pkgname"]
-            $rpName = $matches["service"]
-            $pkgId = "sdk/$rpName/$packageName"
+            $serviceDir = $matches["service"]
+            $pkgId = "sdk/$serviceDir/$packageName"
+            $sdkType = "mgmt"
         }
         else
         {
             $packageName = $matches["pkgname"]
             $pkgId = "sdk/$packageName"
+            $sdkType = "client"
         }
     }
+    return $pkgId, $serviceDir, $sdkType
+}
+
+# rewrite from artifact-metadata-parsing.ps1 used in RetrievePackages for fetch go single module info
+function Get-go-PackageInfoFromPackageFile ($pkg, $workingDirectory)
+{
+    $workFolder = $pkg.Directory
+    $releaseNotes = ""
 
-    $pkgVersion = Get-Version $workFolder
+    $packageProperties = (Get-AllPackageInfoFromRepo -pkgDirectory $workFolder)[0]
```

Comment:
How sure are we that `[0]` is always the correct one here?

Reply:
It should return only one result, i.e. one `go.mod` file